1 game board

Barbirix opened the pull request:
Print a white board divided by blue lines into a 16x9 grid.

Commit:
Background board is displayed (16x9)

Whiteboard with blue lines (16 x 9 square grid) is displayed on the 
screen. All values for window/board size are hardcoded. Mouse 
interaction is not added yet.

Commit:
Calculate board vertices based on window dimensions and board size.

`calculate_board_vertices` is added. Now the board will be displayed for 
different settings: window and board size.
Unittest - only successful scenario is written.

Commit:
Draw cells as separate objects

Every cell is a separate object with a position in a bottom-left corner. 
Board and all cells are drawing in one batch for better performance

Commit:
Estetic change: changing funcion name

Commit:
Cosmetic changes for PEP8 requirements

Commit:
Calculate cell size based on numbers of columns and rows

Cell size will be calculated based on board dimensions and given grid. 
Cell size must be a square and board must be evenly divided by cells.

blazej-jastrzebski-tooploox commented on game_board.py:
```diff
@@ -0,0 +1,99 @@
+
+
+class Cell:
+    def __init__(self, position_x, position_y, cell_size, state):
```
Not necessary, but it's a good idea to add a docstring or a comment with expected data types and 'units' - something like:

```
"""
position_x - int - pixel
position_y - int - pixel
cell_size - int - pixel
state - (No idea what should be put here, what are the possible values and how it will be used and it's not evident in the code, so a comment is even more of a good idea).
```

Barbirix replied:
Added

blazej-jastrzebski-tooploox commented on game_board.py:
```diff
@@ -0,0 +1,99 @@
+                                         0, 0, 255,
+                                         0, 0, 255,
+                                         0, 0, 255,
+                                         0, 0, 255)))
```
The following comment is related to all of your code. The way you write is PEP-8 compliant; it is unfortunately also the worst from the cognitive and practical sense.

Cognitively - I can't find the article now, but basically "the further in the code line, the less understandable the information is" - even including just indented lines.

Practically: we already are using the prohibitive 80 (79, really) char limit; if you break the line like here, you have some 40 chars remaining. If you add a single indentation level, you have 68 chars remaining.

I strongly suggest using the following indentation style:

```
vertex_list = batch.add(
    5, 
    pyglet.gl.GL_LINE_STRIP, 
    None,
    ('v2i', (
        left, bottom, right, bottom, right, top, left, top, left, bottom)),
    ('c3B', (0, 0, 255, 0, 0, 255, 0, 0, 255, 0, 0, 255, 0, 0, 255))
)
```

Barbirix replied:
Changed

blazej-jastrzebski-tooploox commented on game_board.py:
```diff
@@ -0,0 +1,99 @@
+
+
+def draw_background(win_width, win_height,
+                    board_width, board_height, columns_num, rows_num):
```
For similar reasons, the suggested style for situations like this one (a lot of arguments) would be:

```
def draw_background(
    win_width, win_height, board_width, board_height, columns_num, rows_num):
```

Barbirix replied:
changed

blazej-jastrzebski-tooploox commented on game_board.py:
```diff
@@ -0,0 +1,99 @@
+            cell = Cell(position_x, position_y, cell_size, 0)
+            cell.add_cell_to_batch(batch)
+            position_x += cell_size
+        position_x = board_left
```
I might be missing something, but don't you wan't to update the position_y as well?

Barbirix replied:
it is updated after batch.add() function

blazej-jastrzebski-tooploox commented on game_board.py:
```diff
@@ -0,0 +1,99 @@
+    position_y = board_bottom
+    for column in range(rows_num):
+        for row in range(columns_num):
+            cell = Cell(position_x, position_y, cell_size, 0)
```
Are you sure that you want to create new Cells every frame? They won't retain their state this way - it would be better to create them in a separate place, add them to a batch and just draw the batch - AFAIK this works with pyglet.

Barbirix replied:
Done with `prepare_batch` function

blazej-jastrzebski-tooploox commented on game_board.py:
```diff
@@ -0,0 +1,99 @@
+        raise ValueError("The Window size must be bigger than the Board size")
+
+if __name__ == '__main__':
+    pyglet.app.run()
```
Since we're not creating a Window object here, I think this pretty much does nothing.

Barbirix replied:
Agree

blazej-jastrzebski-tooploox commented on requirements.txt:
```diff
@@ -1 +1,2 @@
 pyglet~=1.3
+git+git://github.com/jorgecarleitao/pyglet-gui@master
```
Nice!

Barbirix replied:
It was added by you on my computer, so indeed Nice! but not my work 😄

blazej-jastrzebski-tooploox reviewed (changes requested):
Nice work, looks decent! Could use a few more comments generally, but the code's pretty self-explanatory so...

Commit:
Move batch creation outside the on_draw() function

Move batch preparation from @window.event -> on_draw() function. This 
way board and cells will be created one time in the beggining(not in 
every frame)

Add also docstrings and make styling changes from PEP-8 to Google.
